apps/weather_status/src/App.vue

```diff
@@ -6,36 +6,56 @@
 <template>
 	<div id="weather-status-menu-item">
 		<NcActions class="weather-status-menu-item__subheader"
-			:default-icon="weatherIcon"
 			:aria-hidden="true"
 			:aria-label="currentWeatherMessage"
 			:menu-name="currentWeatherMessage">
+			<template #icon>
+				<NcLoadingIcon v-if="loading" />
+				<img v-else :src="weatherIconUrl" class="weather-image">
+			</template>
 			<NcActionText v-if="gotWeather"
-				:aria-hidden="true"
-				:icon="futureWeatherIcon">
+				:aria-hidden="true">
+				<template #icon>
+					<NcLoadingIcon v-if="loading" />
+					<div v-else class="weather-action-image-container">
+						<img :src="futureWeatherIconUrl" class="weather-image">
+					</div>
+				</template>
 				{{ forecastMessage }}
 			</NcActionText>
 			<NcActionLink v-if="gotWeather"
-				icon="icon-address"
 				target="_blank"
 				:aria-hidden="true"
 				:href="weatherLinkTarget"
 				:close-after-click="true">
+				<template #icon>
+					<NcIconSvgWrapper name="MapMarker"
+						:svg="mapMarkerSvg"
+						:size="20" />
+				</template>
 				{{ locationText }}
 			</NcActionLink>
 			<NcActionButton v-if="gotWeather"
 				:aria-hidden="true"
 				@click="onAddRemoveFavoriteClick">
 				<template #icon>
-					<component :is="addRemoveFavoriteIcon" :size="20" class="favorite-color" />
+					<NcIconSvgWrapper name="Star"
+						:svg="addRemoveFavoriteSvg"
+						:size="20"
+						class="favorite-color" />
 				</template>
 				{{ addRemoveFavoriteText }}
 			</NcActionButton>
 			<NcActionSeparator v-if="address && !errorMessage" />
-			<NcActionButton icon="icon-crosshair"
+			<NcActionButton
 				:close-after-click="true"
```

Comment:
```suggestion
			<NcActionButton :close-after-click="true"
```

(Sadly) our code style requires this